src/components/button/button.type.d.ts

```diff
@@ -0,0 +1,8 @@
+
+export type ButtonTypes =
+  | "primary"
+  | "secondary"
+  | "tertiary"
+  | "dashed"
+  | "destructive"
+  | "darkBackground";
```

Comment:
this could be placed in the main index file

Reply:
Yup this is just couple of lines, moved it into `button.d.ts`